src/actions/collapseContext.ts

```diff
@@ -78,14 +109,26 @@ const collapseContext = (state: State, { at }: Options) => {
       newValue: createId(), // unique value
       path: simplePath,
     }),
+
     // outdent each child
-    ...children.map((child, i) =>
-      moveThought({
+    ...metaChildren.map((child, i) => {
+      const newRank = metaRankStart + metaRankIncrement * i
+      return moveThought({
         oldPath: appendToPath(simplePath, child.id),
         newPath: appendToPath(parentOf(simplePath), child.id),
-        newRank: rankStart + rankIncrement * i,
-      }),
-    ),
+        newRank,
+      })
+    }),
+    ...nonMetaChildren.map((child, i) => {
+      const newRank = canInsertContinuously
+        ? rankStart + rankIncrement * (i + metaChildren.length)
+        : rankStart + rankIncrement * (i + 1)
+      return moveThought({
+        oldPath: appendToPath(simplePath, child.id),
+        newPath: appendToPath(parentOf(simplePath), child.id),
+        newRank,
+      })
```

Comment:
If you need access to the updated state, you can use the uncurried version of `moveThought`.

Instead of:

```ts
// moveThought is a curried function, so passing only one argument instead of the required two will partially 
// apply the function and return a unary reducer to reducerFlow:
moveThought({
  oldPath: ...,
  newPath: ...,
  newRank,
})
```

Do this:

```ts
state => moveThought(state, {
  // now you have access to the new state
  oldPath: ...,
  newPath: ...,
  newRank,
})
```